frontend/public/redux.ts

```diff
@@ -33,17 +36,59 @@ export type RootState = {
   [featureReducerName]: FeatureState;
   [monitoringReducerName]: MonitoringState;
   dashboards: DashboardsState;
+  plugins?: {
+    [namespace: string]: any;
+  };
 };
 
-const reducers = combineReducers<RootState>({
+const baseReducers = Object.freeze({
   k8s: k8sReducers, // data
   UI: UIReducers,
   [featureReducerName]: featureReducer,
   [monitoringReducerName]: monitoringReducer,
   dashboards: dashboardsReducer,
 });
 
-const store = createStore(reducers, {}, composeEnhancers(applyMiddleware(thunk)));
+const store = createStore(
+  combineReducers<RootState>(baseReducers),
+  {},
+  composeEnhancers(applyMiddleware(thunk)),
+);
+
+const addPluginListener = () => {
+  const getReduxFlagsObject = () => {
+    const requiredFlags = registry.getRequiredFlags([isReduxReducer]);
+    const featureState = store.getState()[featureReducerName];
+    return featureState ? _.pick(featureState.toObject(), requiredFlags) : null;
+  };
+
+  let flagsObject = getReduxFlagsObject();
+
+  store.subscribe(() => {
+    const currentFlagsObject = getReduxFlagsObject();
+
+    if (JSON.stringify(flagsObject) !== JSON.stringify(currentFlagsObject)) {
```

Comment:
I think this is probably OK, but it relies on the order of keys in `JSON.stringify` to be stable. It's not guaranteed, but I believe it happens in practice.

Reply:
> We should be mindful of performance here as this check will run a lot.

:+1:

Every time the Console Redux state changes, the new listener gets called, so we narrow the data-to-compare (`getReduxFlagsObject`) down to flags required by _any_ `ReduxReducer` extensions.

```ts
const getReduxFlagsObject = () => {
  const requiredFlags = registry.getRequiredFlags([isReduxReducer]);
  const featureState = store.getState()[featureReducerName]; // <= all flags
  return featureState ? _.pick(featureState.toObject(), requiredFlags) : null; <= narrow flags
};
```

With current code, there is 1x instance of `ReduxReducer` extension (contributed by KubeVirt plugin) gated by the `KUBEVIRT` flag, so `getReduxFlagsObject` will return an object with _only_ `KUBEVIRT` key, at the moment.

> I think flags is a simple flat object, so deep equals is OK. I wonder if we can get away with `===` or is it a different object every time?

It's a different object upon every `getReduxFlagsObject` call due to `_.pick`.